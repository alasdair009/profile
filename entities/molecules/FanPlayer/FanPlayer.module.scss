@use "../../../app/styles/animation";
@use "../../../app/styles/colors";

@keyframes rotate {
  0% {
    background-color: colors.$colorRedHeat;
    transform: translate(-50%, -50%);
  }
  25% {
    background-color: colors.$colorGreenGrass;
  }
  50% {
    background-color: colors.$colorBlueSea;
    transform: translate(-50%, -50%) rotate(360deg);
  }
  75% {
    background-color: colors.$colorblackEvil;
  }
  100% {
    background-color: colors.$colorRedHeat;
    transform: translate(-50%, -50%) rotate(720deg);
  }
}

.root {
  align-items: center;
  display: flex;
  height: 100vh;
  justify-content: center;
  position: relative;
  width: 100vw;

  &::before {
    animation: rotate animation.animationDuration(6) infinite linear;
    aspect-ratio: 1;
    background: colors.$colorGreenGrass;
    mask: url(./assets/fan.svg) no-repeat center;
    max-height: 100%;
    max-width: 100%;
    content: "";
    left: 50%;
    opacity: 0.5;
    position: absolute;
    top: 50%;
    transform: translate(-50%, -50%);
    width: 100%;
  }
}
