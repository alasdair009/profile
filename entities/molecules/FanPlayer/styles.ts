"use client"
import styled, {keyframes} from "styled-components";
import fanSvg from "./assets/fan.svg";
import {colors} from "@/entities";

const rotate = keyframes`
    0% {
        background-color: ${colors.redHeat};
        transform: translate(-50%, -50%);
    }
    25% {
        background-color: ${colors.greenGrass};
    }
    50% {
        background-color: ${colors.blueSea};
        transform: translate(-50%, -50%) rotate(360deg);
    }
    75% {
        background-color: ${colors.blackEvil};
    }
    100% {
        background-color: ${colors.redHeat};
        transform: translate(-50%, -50%) rotate(720deg);
    }
`;
export const Root = styled.div`
    align-items: center;
    display: flex;
    height: 100vh;
    justify-content: center;
    position: relative;
    width: 100vw;
    
    &::before {
        animation: ${rotate} 6s infinite linear;
        aspect-ratio: 1;
        background: ${colors.greenGrass};
        mask: url(${fanSvg.src}) no-repeat center;
        max-height: 100%;
        max-width: 100%;
        content: "";
        left: 50%;
        opacity: 0.5;
        position: absolute;
        top: 50%;
        transform: translate(-50%, -50%);
        width: 100%;
    }
`;