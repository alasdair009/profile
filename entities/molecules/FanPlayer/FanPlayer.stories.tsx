import { Meta, StoryObj } from "@storybook/react";
import { FanPlayer } from "./FanPlayer";

const meta: Meta<typeof FanPlayer> = {
    component: FanPlayer,
};
export default meta;

export const Default: StoryObj<typeof FanPlayer> = {
};
