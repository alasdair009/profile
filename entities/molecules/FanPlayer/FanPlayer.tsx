import { Root } from "./styles";
import { HTMLAttributes } from "react";
import { Audio } from "@/entities";

/**
 * Animated fan that can play a deep turbine noise.
 */
export function FanPlayer({ ...rest }: HTMLAttributes<HTMLDivElement>) {
  return (
    <Root {...rest}>
      <Audio
        src={
          "https://cdn.pixabay.com/download/audio/2022/03/09/audio_03be2a739e.mp3?filename=an-oscillating-fan-23785.mp3"
        }
        autoPlay={true}
      />
    </Root>
  );
}
