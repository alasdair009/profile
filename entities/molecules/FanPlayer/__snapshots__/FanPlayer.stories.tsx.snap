// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`molecules/FanPlayer Default play-test 1`] = `
<div data-testid="FanPlayer"
     class="sc-kGqRax hEJeCg"
>
  <audio controls
         autoplay
         loop
         data-testid="Audio"
         class="sc-braxZu kidHpR"
  >
    <source src="https://cdn.pixabay.com/download/audio/2022/03/09/audio_03be2a739e.mp3?filename=an-oscillating-fan-23785.mp3"
            type="audio/mpeg"
            data-testid="AudioSource"
    >
  </audio>
</div>
`;
